lib/utils/normalizers.ts

```diff
@@ -1,16 +1,78 @@
-import * as url from "url";
-import UrlValueParser = require("url-value-parser");
+import * as url from 'url';
+import UrlValueParser = require('url-value-parser');
 
-export function normalizePath(originalUrl: string, extraMasks: Array<RegExp>, placeholder: string) {
+export function normalizePath(
+  originalUrl: string,
+  extraMasks: Array<RegExp>,
+  placeholder: string,
+) {
   const { pathname } = url.parse(originalUrl);
   const urlParser = new UrlValueParser({ extraMasks });
   return urlParser.replacePathValues(pathname, placeholder);
 }
 
 export function normalizeStatusCode(statusCode: number): string {
-  if (statusCode >= 200 && statusCode < 300) return "2XX";
-  if (statusCode >= 300 && statusCode < 400) return "3XX";
-  if (statusCode >= 400 && statusCode < 500) return "4XX";
+  if (statusCode >= 200 && statusCode < 300) return '2XX';
+  if (statusCode >= 300 && statusCode < 400) return '3XX';
+  if (statusCode >= 400 && statusCode < 500) return '4XX';
 
-  return "5XX";
+  return '5XX';
+}
+
+export function normalizeRoute(req): string {
+  let normalizedRoutePath = req.baseUrl;
+  let endpointRoutePath = req.route.path;
+
+  if (req.route) {
+    if (endpointRoutePath == '/*') {
+      // that means the requested route doesn't exist
+      return 'unknown';
+    }
+
+    if (endpointRoutePath !== '/') {
+      normalizedRoutePath = normalizedRoutePath
+        ? normalizedRoutePath + endpointRoutePath
+        : endpointRoutePath;
+    }
+
+    let originalUrl = req.originalUrl;
+
+    let lastUrlChar = originalUrl.substr(originalUrl.length - 1);
+    // remove redundant trailing slash
+    if (lastUrlChar === '/') {
+      originalUrl = originalUrl.substr(0, originalUrl.length - 1);
+    }
+
+    if (
+      !normalizedRoutePath ||
+      normalizedRoutePath === '' ||
+      typeof normalizedRoutePath !== 'string'
+    ) {
+      // normalizedRoutePath = originalUrl.split('?')[0];
+      // in case the requested route doesn't exist
+      return 'unknown';
+    } else {
+      const splittedRoute = normalizedRoutePath.split('/');
+      const splittedUrl = originalUrl.split('?')[0].split('/');
+      const routeIndex = splittedUrl.length - splittedRoute.length + 1;
+      const baseUrl = splittedUrl.slice(0, routeIndex).join('/');
+      normalizedRoutePath = baseUrl + normalizedRoutePath;
```

Comment:
You removed the query params but as you can see in the original code - it was configurable - so I suggest to return it and have it off by default but allow to set it:

see `this.setupOptions.includeQueryParams` 

```
if (this.setupOptions.includeQueryParams === true && Object.keys(req.query).length > 0) {
                route = `${route}?${Object.keys(req.query).sort().map((queryParam) => `${queryParam}=<?>`).join('&')}`;
            }
```

Reply:
Can I pass it to the normalize method? 
`export function normalizeRoute(req, includeQueryParams = false)`